emr-user-role-mapper-application/src/test/com/amazon/aws/emr/integration/defaultmapper/DefaultMappingProviderImplIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+
+  static private void createAwsResources() {
+    // This bucket is used to read the mapping
+    S3Utils.createBucket(TEST_CFG_BUCKET);
+    // Upload a test object in the test bucket
+    S3Utils.uploadObject(TEST_CFG_BUCKET, TEST_CFG_OBJECT, TEST_S3_OBJECT_CONTENTS);
+    // Find the current AWS principal account
+    String loggedPrincipalAccount = STSUtils.getLoggedUserAccount();
+    // Now create the Role to be used in the Mapping
+    createMappingRole(loggedPrincipalAccount);
+    // Finally create the mapper S3 bucket, and the mapping file
+    uploadDefaultImplMapping();
+  }
+
+  static private void uploadDefaultImplMapping() {
+    S3Utils.createBucket(IntegrationTestBase.DEFAULT_MAPPER_IMPL_BUCKET);
+    String defaultImplMappingJson = defaultImplMappingJsonTemplate
+        .replaceFirst(USER_TO_CHANGE, user)
+        .replaceFirst(USER_ROLE_TO_CHANGE, testRoleArn);
+    S3Utils.uploadObject(IntegrationTestBase.DEFAULT_MAPPER_IMPL_BUCKET,
+        IntegrationTestBase.DEFAULT_MAPPER_IMPL_MAPPING, defaultImplMappingJson);
+  }
+
+  static private void createMappingRole(String awsAccount) {
+    // Replace the principal in the template
+    String roleDoc = rolePolicyDocumentTemplate.replaceFirst(AWS_ACCOUNT_TO_CHANGE, awsAccount);
+    Policy policy = IAMUtils.createPolicy(awsAccount, jsonPolicyDocument);
+    testPolicyArn = policy.getArn();
+    // Now create the role
+    testRoleName = TEST_ROLE_PREFIX + "-" + policy.hashCode() + "-" + roleDoc.hashCode();
+    Role role = IAMUtils.createRole(testRoleName, roleDoc);
+    IAMUtils.attachPolicyToRole(role.getRoleName(), policy.getArn());
+    testRoleArn = role.getArn();
+  }
+
+  @BeforeClass
+  static public void setUp() throws Exception {
+    if (!isOsSupported()) {
+      log.warn("This OS is not supported for integration tests");
+      return;
+    }
+    user = System.getProperty("user.name");
+    createAwsResources();
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    context.setContextPath("/");
+    jettyServer = new Server(pool);
+    jettyServer.setHandler(context);
+
+    ServerConnector httpConnector = new ServerConnector(jettyServer);
+    httpConnector.setPort(TEST_PORT);
+    jettyServer.addConnector(httpConnector);
+
+    ServletHolder jerseyServlet = context.addServlet(ServletContainer.class, "/*");
+    jerseyServlet.setInitOrder(0);
+
+    // Tells the Jersey Servlet which REST service/class to load.
+    jerseyServlet.setInitParameter("jersey.config.server.provider.packages", "com.amazon.emr.api");
+    jerseyServlet
+        .setInitParameter("javax.ws.rs.Application",
+            DefaultProviderImplIntegrationApplication.class.getName());
+
+    log.debug("Starting the Jetty server");
+    jettyServer.start();
+    // AWS Role creation is eventually consistent.
+    // TODO - Fine tune this.
+    Thread.sleep(10 * 1000);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
```

Comment:
don't we need to clean up the uploaded s3 contents as well?

Reply:
Done.